kroxylicious-operator/src/main/java/io/kroxylicious/operator/ProxyDeployment.java

```diff
@@ -0,0 +1,113 @@
+import io.fabric8.kubernetes.api.model.apps.DeploymentBuilder;
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+import io.javaoperatorsdk.operator.processing.dependent.kubernetes.CRUDKubernetesDependentResource;
+import io.javaoperatorsdk.operator.processing.dependent.kubernetes.KubernetesDependent;
+
+import io.kroxylicious.crapi.v1alpha1.KafkaProxy;
+
+/**
+ * The Kube {@code Deployment} for the proxy
+ */
+@KubernetesDependent
+public class ProxyDeployment
+        extends CRUDKubernetesDependentResource<Deployment, KafkaProxy> {
+
+    public static final String CONFIG_VOLUME = "config-volume";
+
+    public ProxyDeployment() {
+        super(Deployment.class);
+    }
+
+    static String deploymentName(KafkaProxy primary) {
+        return primary.getMetadata().getName();
+    }
+
+    @Override
+    protected Deployment desired(KafkaProxy primary,
+                                 Context<KafkaProxy> context) {
+        String configPathInContainer = "/opt/kroxylicious/config/config.yaml";
+        return new DeploymentBuilder()
+                .editOrNewMetadata()
+                .withName(deploymentName(primary))
+                .withNamespace(primary.getMetadata().getNamespace())
+                .withLabels(deploymentLabels())
+                .endMetadata()
+                .editOrNewSpec()
+                .withReplicas(1)
+                .editOrNewSelector()
+                .withMatchLabels(deploymentSelector())
+                .endSelector()
+                .withTemplate(podTemplate(primary, configPathInContainer))
+                .endSpec()
+                .build();
+    }
+
+    private static Map<String, String> deploymentSelector() {
+        return Map.of("app", "kroxylicious");
+    }
+
+    private static Map<String, String> deploymentLabels() {
+        return Map.of("app", "kroxylicious");
+    }
+
+    static Map<String, String> podLabels() {
+        return Map.of("app", "kroxylicious");
+    }
+
+    private PodTemplateSpec podTemplate(KafkaProxy primary, String configPathInContainer) {
+        return new PodTemplateSpecBuilder()
+                .editOrNewMetadata()
+                .withLabels(podLabels())
+                .endMetadata()
+                .editOrNewSpec()
+                .withContainers(proxyContainer(primary, configPathInContainer))
+                .addNewVolume()
+                .withName(CONFIG_VOLUME)
+                .withNewSecret()
+                .withSecretName(ProxyConfigSecret.secretName(primary))
+                .endSecret()
+                .endVolume()
+                .endSpec()
+                .build();
+    }
+
+    private static Container proxyContainer(KafkaProxy primary, String configPathInContainer) {
+        var containerBuilder = new ContainerBuilder()
+                .withName("kroxylicious")
+                .withImage("quay.io/kroxylicious/kroxylicious:0.9.0-SNAPSHOT")
+                .withArgs("--config", configPathInContainer);
```

Comment:
Does anyone else think it's a bit weird to be providing arguments without saying what the command that's consuming them is? I know it's defined by the image itself, but that feels a bit _too_ implicit to me. There are a few alternatives:

* The image defines the full command line (i.e. the config file must be in a specific hard-coded location), which the operator assumes, and mounts the `Secret` there.
* The image uses an env var for the location of the config file, which the operator supplies.
* The operator defines the full command line (i.e. it assumes the location of the binary in the image)